jax/_src/typing/dtypes.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, Literal, Protocol, Union
+import numpy as np
+import ml_dtypes
+
+DType = np.dtype
+
+# TODO(jakevdp, froystig): make ExtendedDType a protocol
+ExtendedDType = Any
```

Comment:
What does this type correspond to?

Reply:
`ExtendedDType` is used for `bint` and `prng_key` dtypes; it's discussed here: https://jax.readthedocs.io/en/latest/jep/9263-typed-keys.html#extended-dtypes